app/integration_action.go

```diff
@@ -171,12 +179,13 @@ func (a *App) DoPostActionWithCookie(postId, actionId, userId, selectedOption st
 	user := ur.Data.(*model.User)
 	upstreamRequest.UserName = user.Username
 
-	tr := <-teamChan
-	if tr.Err != nil {
+	tr, ok := <-teamChan
+	if ok && tr.Err != nil {
 		return "", tr.Err
+	} else if ok {
+		team := tr.Data.(*model.Team)
+		upstreamRequest.TeamName = team.Name
 	}
```

Comment:
I think we can make the code a bit cleaner, if we drop the double check of `ok` and nest the `if`:
```go
	tr, ok := <-teamChan
	if ok {
		if tr.Err != nil {
			return "", tr.Err
		}

		team := tr.Data.(*model.Team)
		upstreamRequest.TeamName = team.Name
	}
```